src/extension.ts

```diff
@@ -145,6 +158,106 @@ async function copyConnectionString(node: IAzureNode<MongoAccountTreeItem | DocD
 	}
 }
 
+async function importDocuments(tree: AzureTreeDataProvider, nodes: vscode.Uri[] | undefined, collectionNode: IAzureParentNode<MongoCollectionTreeItem | DocDBCollectionTreeItem> | undefined): Promise<void> {
+	if (!nodes) {
+		nodes = await askForDocuments();
+	}
+	const documents = await parseDocumentsForErrors(nodes);
+
+	if (!collectionNode) {
+		collectionNode = <IAzureParentNode<MongoCollectionTreeItem | DocDBCollectionTreeItem>>await tree.showNodePicker([MongoCollectionTreeItem.contextValue, DocDBCollectionTreeItem.contextValue]);
+	}
+	let result: string;
+	if (collectionNode.treeItem instanceof MongoCollectionTreeItem) {
+		const collectionTreeItem = <MongoCollectionTreeItem>collectionNode.treeItem;
+		//tslint:disable:no-non-null-assertion
+		result = await collectionTreeItem!.executeCommand('insertMany', [JSON.stringify(documents)]);
+	} else {
+		result = await insertDocumentsIntoDocdb(<IAzureParentNode<DocDBCollectionTreeItem>>collectionNode, documents, nodes);
+	}
+	await collectionNode.refresh();
+	await vscode.window.showInformationMessage(result);
+}
+
+async function askForDocuments(): Promise<vscode.Uri[]> {
+	let files: vscode.Uri[] = await vscode.workspace.findFiles("*.json");
+	let jsonDocuments: (vscode.QuickPickItem & { uri: vscode.Uri })[] = [];
+	let items: (vscode.QuickPickItem & { uri: vscode.Uri })[] = files.map(file => {
+		return { uri: file, label: vscode.workspace.asRelativePath(file) };
+	});
+	let pickAgain: string = "Pick again";
+	let discontinue = "Discontiue import";
+	while (!jsonDocuments.length) {
+		jsonDocuments = await ext.ui.showQuickPick(items, { canPickMany: true, placeHolder: "Choose a document to upload. Hit Escape to Cancel" });
+		if (!jsonDocuments.length) {
+			let action: string = await vscode.window.showWarningMessage("No document picked. Want to pick again?", pickAgain, discontinue);
+			if (action === discontinue) {
+				throw new UserCancelledError();
+			}
+		}
+	}
+	return jsonDocuments.map(choice => choice.uri);
+}
+
+// tslint:disable-next-line:no-any
+async function parseDocumentsForErrors(nodes: vscode.Uri[]): Promise<any[]> {
+	const parseResult = await parseDocuments(nodes);
+	const documents = parseResult[0];
+	const errors: string[] = parseResult[1];
+	if (errors.length > 0) {
+		ext.outputChannel.show();
+		throw new Error(`Errors found in the following documents: ${errors.join(',')}.\nPlease fix these and try again.`);
+	}
+	return documents;
+}
+
+// tslint:disable-next-line:no-any
+async function parseDocuments(nodes: vscode.Uri[]): Promise<[any[], string[]]> {
+	let documents = [];
+	let errors = {};
+	for (let node of nodes) {
+		const document = (await vscode.workspace.openTextDocument(node));
+		const text = document.getText();
```

Comment:
I think this feature has to account for the fact that users might import a lot of files. I don't think reading the files through VS Code is the most efficient (although I haven't done any testing). I would prefer if you do `fse.readFile`